packages/web-config-server/src/apiV1/dataBuilders/generic/percentage/percentagesOfValueCountsPerPeriod.js

```diff
@@ -88,28 +88,19 @@ class BaseBuilder extends PercentagesOfValueCountsBuilder {
   getDataClassesWithAnalytics = async analytics => {
     if (this.config.isProjectReport) {
       const dataWithCountries = await mapAnalyticsToCountries(this.models, analytics);
-      const dataByCountry = groupBy(dataWithCountries, result => result.organisationUnit);
-      // Only one data class is supported for country data classes
-      const baseDataClass = Object.values(this.config.dataClasses)[0];
-      const countryCodesToName = {};
-      const countryCodesToNamePromises = Object.entries(dataByCountry).map(
-        async ([countryCode]) => {
-          const country = await this.models.entity.findOne({ code: countryCode });
-          return { [countryCode]: country.name };
-        },
-        {},
-      );
-      (await Promise.all(countryCodesToNamePromises)).forEach(country =>
-        Object.assign(countryCodesToName, country),
-      );
+      const dataByCountryCode = groupBy(dataWithCountries, 'organisationUnit');
+      const countries = await this.models.entity.find({ code: Object.keys(dataByCountryCode) });
+      const countryCodeToName = reduceToDictionary(countries, 'code', 'name');
 
-      return Object.entries(dataByCountry).reduce(
-        (result, [code, analyticsForCountry]) => ({
-          ...result,
-          [countryCodesToName[code]]: { ...baseDataClass, analytics: analyticsForCountry },
-        }),
-        {},
+      // Only one data class is supported for country data classes
+      const [baseDataClass] = Object.values(this.config.dataClasses);
+      const dataClasses = Object.fromEntries(
+        Object.entries(dataByCountryCode).map(([code, analyticsForCountry]) => [
+          countryCodeToName[code],
+          { ...baseDataClass, analytics: analyticsForCountry },
+        ]),
       );
+      return sortFields(dataClasses);
```

Comment:
Using `sortFields` results in a deterministic sort order (keys are inserted alphabetically), which in turns results in deterministic response data for this report, which addresses flakiness during e2e tests

Reply:
In other words, what matters specifically for e2e tests is not that the fields are ordered alphabetically, but that the same input produces the same output regardless of order